### DISCUSSION TITLE
RESTEasy Reactive for quarkus camel

### DISCUSSION THREAD

**Discussion by EricVincent3** [quarkusio/quarkus#33575] [Q&A] [answered]
hiii, i tried to make my quarkus camel become a reactive camel using RESTEasy Reactive. i added this dependency
<dependency>
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-resteasy-reactive</artifactId>
</dependency>
the problem is, my camel is hitting to a certain URL, before i use netty-http
like this .to(netty-http:xxxx)

but when i try to use resteasy-reactive
.to(resteasy-reactive:xxxxx)

i keep getting error of url not found
No endpoint could be found for: resteasy-reactive://xxxx, please check your classpath contains the needed Camel component jar.

the quarkus version i use is 2.7.4.Final. from information i got, that it is should've been able to use resteasy-reactive

did i miss something or what? thanks in advance

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
cc @quarkusio/camel

**Comment by jamesnetherton** [comment, marked as answer]
There is no `resteasy-reactive` component extension for Camel. So you'll need to use one of the supported HTTP producers like `netty-http`, `vertx-http` or `http`.

You can find the list of supported Camel Quarkus extensions here:

https://camel.apache.org/camel-quarkus/3.0.x/reference/index.html

**Comment by ggrebert** [comment]
Maybe, you can use `camel-quarkus-direct`

```java
from("direct:foo").process(
  exchange -> {
    var toto = exchange.getIn().getBody(Foo.class);
    exchange.getIn().setBody(new Bar());
 });
```

and then:

```java
@Inject CamelContext camel;

@POST
public Uni<Bar> foo(Foo foo) {
  var cs = camel.createProducerTemplate().asyncRequestBody("direct:foo", foo);
  return Uni.createFrom().completionStage(cs);
}
```
